src/settings/variableSubstitution.ts

```diff
@@ -99,11 +100,9 @@ const VARIABLE_SUBSTITUTIONS: VariableSubstitution[] = [
 export function getVariableSubstitutedAssociations(associations: XMLFileAssociation[]): XMLFileAssociation[] {
 
   // Collect properties needed to resolve variables
-  const currentFile: TextDocument = (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.languageId === 'xml') ?  window.activeTextEditor.document : undefined;
-  const currentFileUri: string = (currentFile && currentFile.uri) ? currentFile.uri.fsPath : undefined;
-  const currentWorkspace: WorkspaceFolder = (currentFile && currentFile.uri) ?  workspace.getWorkspaceFolder(currentFile.uri) : undefined;
-  const currentWorkspaceUri: string = (currentWorkspace && currentWorkspace.uri.fsPath)
-    || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath);
+  const currentFile = (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.languageId === 'xml') ? window.activeTextEditor.document : undefined;
+  const currentFileUri: string = getFilePath(currentFile);
+  const currentWorkspaceUri: string = getWorkspaceUri(currentFile);
```

Comment:
Replace with

`const currentWorkspaceUri= getWorkspaceUri(currentFile);`

Reply:
And line 25, replace with

`      newVal = settingVariable.substituteString(newVal, currentFileUri, currentWorkspaceUri && currentWorkspaceUri.toString());`